interfaces/mount/spec.go

```diff
@@ -150,6 +150,29 @@ func (spec *Specification) AddExtraLayouts(layouts []snap.Layout) {
 	}
 }
 
+// createEnsureDirMountEntry creates a mount entry from the given ensure directory spec
+// that instruct snap-update-ns to create missing directories if required.
+func createEnsureDirMountEntry(ensureDirSpec *interfaces.EnsureDirSpec) osutil.MountEntry {
```

Comment:
the apparmor code is cleaning the dirs but this code is not? should they be consistent?

Reply:
I have now removed the cleaning from apparmor counterpart. 

The BeforePreparePlug checks covers everything including clean paths and prefix $HOME etc.
The existing pattern seems to be to trust such checks and not re-validate structures passed, thus typically avoiding errors mid emits.

I suppose it is not ideal for spec functions to rely that all interfaces construct its inputs correctly, they should arguably be more independent and have a catch all. 

Let me know your preference, but for now i think this follows the existing pattern.